File: R/matSignes.R
matSignes <- function(k) {
  # signes <- matSignes(k);
  # G??n??re une matrice de signes selon la valeur de k
  if(!any(k == 3:5))   stop("matSignes(k) pr??vu seulement pour k entre 3 et 5")
  
  MAT <- list(list(),list(),
              matrix(c(1, 1, -1,
                       1, -1, 1,
                       -1, 1, 1), nrow = 3, byrow = TRUE),
              `4` = matrix(c(1, 1, -1, -1,
                             1, -1, 1, -1,
                             1, -1, -1, 1,
                             -1, 1, 1, -1,
                             -1, 1, -1, 1,
                             -1, -1, 1, 1), nrow = 6, byrow = TRUE),
              `5` = matrix(c(1, 1, -1, -1, -1,
                             1, -1, 1, -1, -1,
                             1, -1, -1, 1, -1,
                             1, -1, -1, -1, 1,
                             -1, 1, 1, -1, -1,
                             -1, 1, -1, 1, -1,
                             -1, 1, -1, -1, 1,
                             -1, -1, 1, 1, -1,
                             -1, -1, 1, -1, 1,
                             -1, -1, -1, 1, 1), nrow = 10, byrow = TRUE))
  
  return(MAT[[k]])
}

File: Matlab/SCRoF_mars25/matSignes.m
function signes=matSignes(k)
% signes=matSignes(k);
switch k
    case 3
        signes=[1 1 -1;1 -1 1;-1 1 1];
    case 4
        signes=[1 1 -1 -1;1 -1 1 -1;1 -1 -1 1;-1 1 1 -1;-1 1 -1 1;-1 -1 1 1];
    case 5
        signes=[1 1 -1 -1 -1;1 -1 1 -1 -1;
                1 -1 -1 1 -1;1 -1 -1 -1 1;
                -1 1 1 -1 -1;-1 1 -1 1 -1;
                -1 1 -1 -1 1;-1 -1 1 1 -1;
                -1 -1 1 -1 1;-1 -1 -1 1 1];
    otherwise
        error('matSignes(k) prévu seulement pour k entre 3 et 5')
end
